### ISSUE TITLE
Accessibility-ready review

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwenty#785]
Default themes need to follow the requirements for the [accessibility-ready tag]
(https://make.wordpress.org/themes/handbook/review/accessibility/required/).

**Important:** The accessibility-ready requirements are **simplified,** all code added to core must conform with the [WCAG 2.0 guidelines at level AA. ](https://www.w3.org/TR/WCAG20/)

https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/

**Note 2**: this will most likely be updated to WCAG 2.1 in the near future so we should have this in mind.
https://www.w3.org/TR/WCAG21/

https://make.wordpress.org/accessibility/handbook/test-for-web-accessibility/

Reviews to complete for the **tag**:

- [ ] Keyboard Navigation
- [x] Controls
- [x] Skip Links
- [x] Forms
- [x] Headings
- [x] ARIA Landmark Roles
- [x] Content Links
- [ ] Repetitive Link Text
- [x] Contrasts
- [x] Images
- [x] Media
- [x] Screen Reader Text
- [x] Not Allowed

**Comment by ntwb** [comment]
If possible it would be great to have Twenty Twenty WCAG 2.1 compatible

**Comment by erikkroes** [comment]
The theme uses several HTML entities like `&larr;` throughout. For example, in pagination. They can be read by screen readers, but are functionally comparable to decorative images. 
Adding `aria-hidden` on the containing element, or adding a `<span>` with `aria-hidden` makes screen readers ignore them.
Fixed in #865

**Comment by carolinan** [comment]
@pattonwebz The starter content does not follow the repetitive link requirement https://make.wordpress.org/themes/handbook/review/accessibility/required/#repetitive-link-text 
for the read more links:

`'<div class="wp-block-button is-style-outline"><a class="wp-block-button__link" href="https://make.wordpress.org/core/2019/09/27/block-editor-theme-related-updates-in-wordpress-5-3/">' . __( 'Read More', 'twentytwenty' ) . '</a></div>',`

**Comment by joyously** [comment]
Is that a core Read More block? Or a button block that doesn't need to be there at all?

**Comment by carolinan** [comment]
It is a button block that needs to be there.

**Comment by carolinan** [comment]
Hi
I held it off because I am not 100% happy with the submenu toggles for the desktop horizontal menu. I still feel that the toggles should be separate from the parent menu item.
I have been unable to work on it.
Other than that and the repetitive links in the starter content I have tested the theme as far as I can.

**Comment by yannickiki** [comment]
@carolinan there is still an issue with contrast ratio (#814) when the secondary color is generated.

**To reproduce the issue:**
Inside the Customizer, in order to generate the secondary color, change the background color to white (#fffff) and change it back to default. Publish the changes.

The generated secondary color (that is used for example on the post meta on the blog index) is not AA compliant. The text color #7E704B on the background color #F5EFE0 has a contrast ratio of 4.25 (required contrast ratio: 4.5).

This also fails for other background colors, it depends on the selected color.

![ratio](https://user-images.githubusercontent.com/54663751/68317082-cf4bf480-00ba-11ea-86d5-783f9a3dff30.png)

**Comment by carolinan** [comment]
The requirements only cover the defaults.

**Comment by joedolson** [comment]
If I understand this correctly, this is a problem with the defaults when they are restored - is that correct? I think that a user should be able to restore the defaults and still have accessible contrasts. Right now, it seems like if you *ever* change the background color - even to experiment - you will be unable to restore to the original look, because of the automatically generated secondary color.

**Comment by carolinan** [comment]
Yes that is correct.

**Comment by yannickiki** [comment]
@carolinan #932 didn't mention accessibility issue that why I thought it was important to mention it.

**Comment by aristath** [comment]
Pushed a PR that should fix the secondary color contrast (see #977)

#932 is a completely different issue and I've been working on it for the past few hours, not an easy fix (if you can even call it a bug)
